internal/security/svid/svid.go

```diff
@@ -0,0 +1,123 @@
+// Package svid provides functions to generate and sign svid.
+package svid
+
+import (
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"strings"
+
+	"math/big"
+	"net/url"
+
+	"time"
+)
+
+// GenRSASVID Generates SVID for user and signs it based on given rsa cert/key
+func GenRSASVID(id string, expireInDays int, signingCert *x509.Certificate, signingKey any) (*rsa.PrivateKey, *x509.Certificate, error) {
+	certSpec, err := populateCertTemplate(id, expireInDays)
+	if err != nil {
+		return nil, nil, err
+	}
+	privKey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return nil, nil, err
+	}
+	certBytes, err := x509.CreateCertificate(rand.Reader, certSpec, signingCert, &privKey.PublicKey, signingKey)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	cert, err := x509.ParseCertificate(certBytes)
+	if err != nil {
+		return nil, nil, err
+	}
+	return privKey, cert, nil
+}
+
+// GenECDSASVID Generates SVID for user and signs it based on given ECDSA cert/key
+func GenECDSASVID(id string, expireInDays int, signingCert *x509.Certificate, signingKey any) (*ecdsa.PrivateKey, *x509.Certificate, error) {
+	certSpec, err := populateCertTemplate(id, expireInDays)
+	if err != nil {
+		return nil, nil, err
+	}
+	curve := elliptic.P256()
```

Comment:
the 'create key' part of this function and it's northern partner are the ONLY parts which are different between these 2 functions.

I'd refactor this to have a createKey which does the proper rsa/ecdsa key creation and then have the cert signing/creation just be all in one function.

Copy/paste code is 1 paste away from problems :(

Reply:
fixed the code, please check